src/main/java/seedu/address/logic/parser/task/TaskSearchCommandParser.java

```diff
@@ -0,0 +1,84 @@
+package seedu.address.logic.parser.task;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DAY;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_MONTH;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_YEAR;
+
+import java.util.HashMap;
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.task.TaskSearchCommand;
+import seedu.address.logic.parser.ArgumentMultimap;
+import seedu.address.logic.parser.ArgumentTokenizer;
+import seedu.address.logic.parser.Parser;
+import seedu.address.logic.parser.Prefix;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.task.TaskSearchPredicate;
+
+
+/**
+ * Parses input arguments and creates a new TaskAddCommand object
+ */
+public class TaskSearchCommandParser implements Parser<TaskSearchCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the TaskAddCommand
+     * and returns a TaskAddCommand object for execution.
```

Comment:
I think can change TaskAddCommand to TaskSearchCommand in the javadoc

Reply:
Resolved. Thanks!